.github/workflows/Dockerfile.yml

```diff
@@ -19,55 +19,69 @@ env:
   # leaving sdk npm version blank defaults to whatever npm has for latest version
   # rather than build from git source, which is fine for ci test build
   JS_STELLAR_SDK_NPM_VERSION:
-  RS_XDR_GIT_REPO: https://github.com/stellar/rs-stellar-xdr
-  RS_XDR_GIT_REF: main
 jobs:
   complete:
     if: always()
     needs: [build]
     runs-on: ubuntu-latest
     steps:
-    - if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
-      run: exit 1
+      - if: contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled')
+        run: exit 1
 
   build:
     runs-on: ubuntu-latest-16-cores
     outputs:
       push_enabled: ${{ env.PUSH_ENABLED }}
     steps:
-    - uses: actions/checkout@v4
-      with:
-        ref: ${{ env.HEAD_SHA }}
-    - name: Build System Test Image
-      run: |
-        make CORE_GIT_REF=${{ env.CORE_GIT_REF }} \
-             CORE_COMPILE_CONFIGURE_FLAGS="${{ env.CORE_COMPILE_CONFIGURE_FLAGS }}" \
-             STELLAR_RPC_GIT_REF=${{ env.STELLAR_RPC_GIT_REF }} \
-             RUST_TOOLCHAIN_VERSION=${{ env.RUST_TOOLCHAIN_VERSION }} \
-             SOROBAN_CLI_GIT_REF=${{ env.SOROBAN_CLI_GIT_REF }} \
-             SYSTEM_TEST_IMAGE=${{ env.PUSH_ENABLED == 'true' && env.IMAGE || 'stellar/system-test:dev' }} \
-             JS_STELLAR_SDK_NPM_VERSION=${{ env.JS_STELLAR_SDK_NPM_VERSION }} \
-             RS_XDR_GIT_REPO=${{ env.RS_XDR_GIT_REPO }} \
-             RS_XDR_GIT_REF=${{ env.RS_XDR_GIT_REF }} \
-             QUICKSTART_GIT_REF=${{ env.QUICKSTART_GIT_REF }} build;
-    - if: ${{ env.PUSH_ENABLED == 'true' }}
-      name: Save Docker Image to file
-      run: |
-        docker save ${{ env.IMAGE }} -o /tmp/image;
-    - if: ${{ env.PUSH_ENABLED == 'true' }}
-      name: Upload System Test Image
-      uses: actions/upload-artifact@v4
-      with:
-        name: image-Dockerfile
-        path: /tmp/image
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ env.HEAD_SHA }}
+      - name: Restore image cache
+        id: image-cache
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/image.tar.zst
+          key: systemtest-image-${{ runner.os }}-${{ github.sha }}
+      
+      - name: Load cached Docker image
+        if: steps.image-cache.outputs.cache-hit == 'true'
+        run: |
+          zstd -d -c /tmp/image.tar.zst | docker load
+      
+      - name: Build System Test Image
+        if: steps.image-cache.outputs.cache-hit != 'true'
+        run: |
+          make SYSTEM_TEST_IMAGE=${{ env.PUSH_ENABLED == 'true' && env.IMAGE || 'stellar/system-test:dev' }} \
+               JS_STELLAR_SDK_NPM_VERSION=${{ env.JS_STELLAR_SDK_NPM_VERSION }} \
+               build;
+
+      - name: Save Docker Image to file
+        if: steps.image-cache.outputs.cache-hit != 'true'
+        run: |
+          docker save ${{ env.PUSH_ENABLED == 'true' && env.IMAGE || 'stellar/system-test:dev' }} | zstd -19 -o /tmp/image.tar.zst
```

Comment:
If you direct this output to .tar, then separately pipe that to zstd, then you don't need to decompress the file in the next step and reduce the number of compression operations.

If you want both to happen at the same time use `docker save ... | tee /tmp/image.tar | ztd ...`